applications/nrf_desktop/doc/config_channel.rst

```diff
@@ -92,184 +92,233 @@ The data format is described by the following table.
-   The dongle would act as a bridge between the host and the configured peripheral.
 
-If a device connected via dongle disconnects, you must discover devices connected via the dongle again to regain ability to configure it.
-The recipient ID that used to be assigned to the device before it disconnected is no longer valid.
-Discovering devices connected via the dongle is also needed to access devices that connected to the dongle after peers were indexed.
+The assigned Recipient ID is valid only during connection time between the remote device and the dongle.
+If a remote device disconnects from the dongle, the recipient discovery must be repeated when device reconnects.
 
 .. tip::
-   See :ref:`nrf_desktop_config_channel_script` for implementation example of host tools.
+   The recipient discovery is not needed if host does not interact with remote devices connected through dongle.
+
+.. _nrf_desktop_config_channel_device_identification:
+
+Device identification
+---------------------
+
+The following requests provide the information about the device hardware.
+Use the Recipient ID linked with the device that is being discovered.
+
+1. Read board name using ``CONFIG_STATUS_GET_BOARD_NAME`` request.
+#. Read hardware ID using ``CONFIG_STATUS_GET_HWID`` request.
+
+The Hardware ID allows to differentiate devices of the same type (having the same board name).
+The tool should note which Recipient ID is associated with the board name and Hardware ID.
+
+.. _nrf_desktop_config_channel_module_discovery:
+
+Module discovery
+----------------
+
+The host can access options provided by the application modules.
+Before the options are accessed the host must identified which modules are available and what options are provided by them.
+Both modules and their options are identified by name, sent as a string during discovery procedure.
+The module of given name is linked with specific value of Module ID.
+The option of given name is linked with specific value of Option ID.
+The Module ID and Option ID associated with module and option name may vary between devices, but do not change between connections to the same device and firmware version.
+
+The module discovery procedure consists of a following steps:
+
+1. Obtain the number of configurable application modules using ``CONFIG_STATUS_GET_MAX_MOD_ID`` request.
+#. Read module descriptor of every application module (iterate Module ID from 0 up to and including max supported Module ID):
+
+   1. The module descriptor is read using fetch request with current Module ID while Option ID is set to ``0``.
+      The request is followed by a set of responses, each containing a string being part of the module descriptor.
+      The end line character (``\n``) indicates the end of the descriptor.
+      The host can stop option discovery procedure at this point as subsequent requests will loop around and repeat descriptor strings.
```

Comment:
We are not sure which descriptor would be sent as first one. Because of that we must wait until we get a duplicated record.

Reply:
We start from 0 but indeed if somebody breaks query earlier it will get out of sync. Shame we have not added the reset request. Anyway this will indeed need to be reworded.

Reply:
We rely on fetching duplicate in Python scripts so we would not start from zero even after a successful discovery. I would not say it's a problem for host tools (we handle it easily in our Python scripts).